Wikipedia/Code/DiffContainerViewController.swift

```diff
@@ -691,3 +880,127 @@ class RevisionAuthorThanksErrorHintVC: HintViewController {
         warningSubtitleLabel.text = nil
     }
 }
+
+extension DiffContainerViewController: DiffToolbarViewDelegate {
+    
+    private func replaceLastAndPush(with viewController: UIViewController) {
+        if var newViewControllers = navigationController?.viewControllers {
+            newViewControllers.removeLast()
+            newViewControllers.append(viewController)
+            navigationController?.setViewControllers(newViewControllers, animated: true)
+        }
+    }
+    
+    func tappedPrevious() {
+        
+        self.navigationController?.delegate = self
```

Comment:
Setting the navigation controller's delegate here (and nil'ing it out elsewhere) breaks the transitions handled by `WMFAppViewController` (explore feed, search) as well as some other functionality it handles as the nav delegate. It seems that since `WMFAppViewController` sets up the nav controller it should remain the delegate. Could the prev/next transitions be added to `ViewControllerTransitionsController` which `WMFAppViewController` uses to set up transitions? Feel free to refactor or adjust `ViewControllerTransitionsController` if you see a better way for it to detect which transition to use

Reply:
Great catch! I totally missed that - this is integrated into `ViewControllerTransitionsController` now, I think that fixes the issues you were seeing.